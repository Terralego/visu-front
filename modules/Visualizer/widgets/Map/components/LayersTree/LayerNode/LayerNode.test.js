import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LayersNode from '.';

it('should render correctly', () => {
  const tree = renderer.create((
    <div>
      <LayersNode
        LayersTree={[]}
      />
      <LayersNode
        LayersTree={[]}
        isActive
      />
    </div>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should open options panel', () => {
  const wrapper = shallow(<LayersNode />);
  const instance = wrapper.instance();
  jest.spyOn(instance, 'setState');
  instance.handleOptionPanel();
  expect(instance.setState).toHaveBeenCalledWith({ isOptionsOpen: true });
  instance.setState.mockClear();
  instance.handleOptionPanel();
  expect(instance.setState).toHaveBeenCalledWith({ isOptionsOpen: false });
});

it('should render options layer', () => {
  const wrapper = shallow(<LayersNode isActive />);
  wrapper.setState({ isOptionsOpen: true });
  expect(wrapper.find('OptionsLayer').length).toBe(1);
})
;