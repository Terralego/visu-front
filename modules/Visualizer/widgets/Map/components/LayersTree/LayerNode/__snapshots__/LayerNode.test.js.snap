// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="bp3-card bp3-elevation-2 layerNode-container"
    style={
      Object {
        "opacity": 0.7,
      }
    }
  >
    <label
      className="bp3-control bp3-switch"
    >
      <input
        type="checkbox"
      />
      <span
        className="bp3-control-indicator"
      />
    </label>
  </div>,
  <div
    className="bp3-card bp3-elevation-2 layerNode-container"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <label
      className="bp3-control bp3-switch"
    >
      <input
        checked={true}
        type="checkbox"
      />
      <span
        className="bp3-control-indicator"
      />
    </label>
  </div>,
]
`;
